launch/launch/launch_service.py

```diff
@@ -185,37 +184,26 @@ def _prepare_run_loop(self):
                 this_task = asyncio.Task.current_task(this_loop)
 
             self.__this_task = this_task
-            # Setup custom signal handlers for SIGINT, SIGTERM and maybe SIGQUIT.
-            sigint_received = False
+            # Setup custom signal handlers for SIGINT & SIGTERM
+            signals_received: dict = {}
 
-            def _on_sigint(signum):
-                nonlocal sigint_received
-                base_msg = 'user interrupted with ctrl-c (SIGINT)'
-                if not sigint_received:
+            def _on_signal(signum):
+                nonlocal signals_received
+                base_msg = 'caught ' + signal.Signals(signum).name
+                if signum not in signals_received or signals_received[signum] is not True:
                     self.__logger.warning(base_msg)
                     ret = self._shutdown(
-                        reason='ctrl-c (SIGINT)', due_to_sigint=True, force_sync=True
+                        reason=base_msg, due_to_sigint=True, force_sync=True
```

Comment:
the name should be changed into `due_to_signal`? cz it is not only with sigint anymore.

```suggestion
                        reason=base_msg, due_to_signal=True, force_sync=True
```

Reply:
It should be more generic yes, but currently the semantics of SIGINT are different to other signals, as when it's combined with the `noninteractive` flag the signal is either forwarded or not. Will require further changes in different files that rely on this variable, looking into that now.

Reply:
Decided to keep the original `due_to_sigint` as for now we're only handling two signals, SIGINT & SIGTERM, so `due_to_sigint=False` unambiguously implies SIGTERM.